Examples/UIExplorer/NavigationExperimental/NavigationCustomStylingExample.js

```diff
@@ -0,0 +1,179 @@
+*/
+'use strict';
+
+const React = require('react-native');
+
+const {
+  Animated,
+  NavigationExperimental,
+  StyleSheet,
+  ScrollView,
+} = React;
+
+const  NavigationExampleRow = require('./NavigationExampleRow');
+
+const  {
+  AnimatedView: NavigationAnimatedView,
+  Card: NavigationCard,
+  Header: NavigationHeader,
+  Reducer: NavigationReducer,
+  RootContainer: NavigationRootContainer,
+} = NavigationExperimental;
+
+const NavigationBasicReducer = NavigationReducer.StackReducer({
+  getPushedReducerForAction: (action) => {
+    if (action.type === 'push') {
+      return (state) => state || {key: action.key};
+    }
+    return null;
+  },
+  getReducerForState: (initialState) => (state) => state || initialState,
+  initialState: {
+    key: 'AnimatedExampleStackKey',
+    index: 0,
+    children: [
+      {key: 'First Route'},
+    ],
+  },
+});
+
+class NavigationCustomStylingExample extends React.Component {
+  componentWillMount() {
+    this._renderNavigation = this._renderNavigation.bind(this);
+    this._renderCard = this._renderCard.bind(this);
+    this._renderScene = this._renderScene.bind(this);
+    this._renderHeader = this._renderHeader.bind(this);
+  }
+  render() {
+    return (
+      <NavigationRootContainer
+        reducer={NavigationBasicReducer}
+        ref={navRootContainer => { this.navRootContainer = navRootContainer; }}
+        persistenceKey="NavigationCustomStylingExampleState"
+        renderNavigation={this._renderNavigation}
+      />
+    );
+  }
+  handleBackAction() {
+    return (
+      this.navRootContainer &&
+      this.navRootContainer.handleNavigation(NavigationRootContainer.getBackAction())
+    );
+  }
+  _renderNavigation(navigationState, onNavigate) {
+    if (!navigationState) {
+      return null;
+    }
+    return (
+      <NavigationAnimatedView
+        navigationState={navigationState}
+        style={styles.animatedView}
+        renderOverlay={this._renderHeader}
+        applyAnimation={(pos, navState) => {
+          Animated.timing(pos, {toValue: navState.index, duration: 500}).start();
+        }}
+        renderScene={this._renderCard}
+      />
+    );
+  }
+
+  _renderHeader(/*NavigationSceneRendererProps*/ props) {
+    return (
+      <NavigationHeader
+        style={styles.header}
+        navigationProps={props}
+        renderTitleComponent={(navigationProps, scene) => {
+          return (
+            <NavigationHeader.Title
+              textStyle={styles.titleText}>
+              {scene.navigationState.key}
+            </NavigationHeader.Title>);
+        }}
+        renderLeftComponent={(navigationProps, scene) => {
+          return (
+            <NavigationHeader.BackButton
+              imageSource={require('./assets/back-icon.png')}
+            />);
+        }}
+      />
+    );
+  }
+
+  _renderCard(/*NavigationSceneRendererProps*/ props) {
+    return (
+      <NavigationCard
+        {...props}
+        style={styles.card}
```

Comment:
FYI

I'm not sure whether you're aware of that this will overwrite the build-in style, which also includes the styles needed for animation.

If you do want to keep the animation, you may do this:

```
_renderCard(/*NavigationSceneRendererProps*/ props) {
  return (
    <NavigationCard
      {...props}
      style={[
        NavigationCardStackStyleInterpolator.forHorizontal(props),
        styles.card,
      ]}
    />
  );
}
```